app/scripts/filters/exists.coffee

```diff
@@ -17,6 +17,8 @@ angular.module('edudashAppFil')
                 for prop, value of input
                     return true;
                 return false
+            if angular.isNumber input && input == NaN
```

Comment:
Haha I read the coffeescript wrong. Here's the JS it compiles to:
```javascript
if (angular.isNumber(input && input === NaN)) {
```

which is interesting and pretty tricky. The right half of the `&&` is always `false`, `angular.isNumber(false)` is false, so it's only true if the left side is truth-y and a number. NaN is false-y which is why this passes the test, but so is zero, and I think `exists 0` would be expected to be `true`.

There is an `isNaN` global JS function, but it has [weird behaviour](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#Examples). So, here is a way that should be reliable and not surprising:

```coffeescript
if (angular.isNumber input) and (input == input)
```
The second part fails for `NaN`. Probably worth a comment.

javascript is just the worst

in case you haven't seen it: [wat](https://www.destroyallsoftware.com/talks/wat)